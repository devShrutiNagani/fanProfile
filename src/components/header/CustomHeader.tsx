import React from 'react';
import {Image, Text, TouchableOpacity, View} from 'react-native';
import {RFValue} from 'react-native-responsive-fontsize';
import createStyles from './Styles';
import {CustomHeaderProps} from '../../constant/interface/custom/Custom';

const CustomHeader: React.FC<CustomHeaderProps> = ({
  firstIcon,
  title,
  titleTextStyle,
  firstIconStyle,
  children,
  backPress,
}) => {
  const styles = createStyles();
  return (
    <View style={styles.container}>
      {firstIcon && (
        <TouchableOpacity
          onPress={backPress}
          style={styles.firstIconContainer}
          testID="back-button">
          <Image
            testID="first-icon"
            style={[styles.firstIcon, firstIconStyle]}
            source={firstIcon}
          />
        </TouchableOpacity>
      )}

      <View style={styles.titleView}>
        {title && (
          <Text
            style={[
              styles.titleText,
              titleTextStyle,
              {marginRight: firstIcon && RFValue(30)},
            ]}
            numberOfLines={1}>
            {title}
          </Text>
        )}
        {children}
      </View>
    </View>
  );
};

export default CustomHeader;
