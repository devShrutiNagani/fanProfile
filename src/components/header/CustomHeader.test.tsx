import React from 'react';
import {render, fireEvent} from '@testing-library/react-native';
import CustomHeader from './CustomHeader';
import {Text} from 'react-native';

describe('CustomHeader Component', () => {
  it('renders the title correctly', () => {
    const {getByText} = render(<CustomHeader title="Test Header" />);
    expect(getByText('Test Header')).toBeTruthy();
  });

  it('renders the first icon when provided', () => {
    const {getByTestId} = render(
      <CustomHeader firstIcon={{uri: 'test-icon'}} />,
    );
    expect(getByTestId('first-icon')).toBeTruthy();
  });

  it('calls backPress function when first icon is pressed', () => {
    const mockBackPress = jest.fn();
    const {getByTestId} = render(
      <CustomHeader firstIcon={{uri: 'test-icon'}} backPress={mockBackPress} />,
    );

    fireEvent.press(getByTestId('first-icon'));
    expect(mockBackPress).toHaveBeenCalledTimes(1);
  });

  it('renders children elements when passed', () => {
    const {getByText} = render(
      <CustomHeader>
        <Text>Child Element</Text>
      </CustomHeader>,
    );

    expect(getByText('Child Element')).toBeTruthy();
  });

  it('matches snapshot', () => {
    const tree = render(
      <CustomHeader title="Snapshot Header" firstIcon={{uri: 'test-icon'}} />,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
