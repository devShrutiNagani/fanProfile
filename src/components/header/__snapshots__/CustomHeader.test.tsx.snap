// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomHeader Component matches snapshot 1`] = `
<View
  style={
    {
      "flex": 1,
      "flexDirection": "row",
      "marginVertical": 20,
    }
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={false}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "opacity": 1,
        "width": 59,
      }
    }
    testID="back-button"
  >
    <Image
      source={
        {
          "uri": "test-icon",
        }
      }
      style={
        [
          {
            "height": 39,
            "resizeMode": "contain",
            "tintColor": "#1D1D1D",
            "width": 39,
          },
          undefined,
        ]
      }
      testID="first-icon"
    />
  </View>
  <View
    style={
      {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        [
          {
            "color": "#1D1D1D",
            "fontFamily": "Montserrat-Medium",
            "fontSize": 35,
            "includeFontPadding": false,
          },
          undefined,
          {
            "marginRight": 59,
          },
        ]
      }
    >
      Snapshot Header
    </Text>
  </View>
</View>
`;
